### PR TITLE
Велиев Эльвин Задача 3 Вариант 29. Выделение ребер на изображении с использованием оператора Собеля

### PR DESCRIPTION
**Pull request by ev1ven**
...

### CODE REVIEW

File: tasks/mpi/veliev_e_sobel_operator/func_tests/main.cpp [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+namespace veliev_e_sobel_operator_mpi {
+std::vector<double> create_random_vector(int size) {
+  std::uniform_real_distribution<double> unif(static_cast<double>(0), static_cast<double>(255));
+  std::random_device rand_dev;
+  std::mt19937 rand_engine(rand_dev());
+  std::vector<double> tmp(size);
+  for (int i = 0; i < size; i++) {
+    tmp[i] = unif(rand_engine);
+  }
```
Comment (qfrff) [inline tasks/mpi/veliev_e_sobel_operator/func_tests/main.cpp:22]: optimize random vector generation

File: tasks/mpi/veliev_e_sobel_operator/perf_tests/main.cpp
Comment (qfrff) [inline tasks/mpi/veliev_e_sobel_operator/perf_tests/main.cpp:1]: also here

File: tasks/seq/veliev_e_sobel_operator/perf_tests/main.cpp
Comment (qfrff) [inline tasks/seq/veliev_e_sobel_operator/perf_tests/main.cpp:1]: and here

File: tasks/seq/veliev_e_sobel_operator/src/ops_seq.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      result[y * w + x] = std::sqrt(gx * gx + gy * gy);
+    }
+  }
+
+  double max_val = *std::max_element(result.begin(), result.end());
+  if (max_val > 0.0) {
+    for (double &val : result) {
+      val /= max_val;
+    }
+  }
```
Comment (qfrff) [inline tasks/seq/veliev_e_sobel_operator/src/ops_seq.cpp:33]: replace the result normalization loop with std::transform:

File: tasks/mpi/veliev_e_sobel_operator/src/ops_mpi.cpp
Comment (qfrff) [inline tasks/mpi/veliev_e_sobel_operator/src/ops_mpi.cpp:1]: and here

File: tasks/seq/veliev_e_sobel_operator/src/ops_seq.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 Nesterov Alexander
+#include "seq/veliev_e_sobel_operator/include/ops_seq.hpp"
+
+std::vector<double> sobel_filter(const std::vector<double> &image_vector, int h, int w) {
```
Comment (Sedova-Olga) [inline tasks/seq/veliev_e_sobel_operator/src/ops_seq.cpp:4]: Cyclic normalization is built into the core filter logic, which reduces readability. Create a separate function to normalize the vector.

**Review by Sedova-Olga — CHANGES_REQUESTED** [review]

File: tasks/seq/veliev_e_sobel_operator/src/ops_seq.cpp (hunk 3 above)
Comment (ev1ven) [reply tasks/seq/veliev_e_sobel_operator/src/ops_seq.cpp]: fixed

File: tasks/mpi/veliev_e_sobel_operator/src/ops_mpi.cpp
Comment (ev1ven) [reply tasks/mpi/veliev_e_sobel_operator/src/ops_mpi.cpp:1]: fixed

File: tasks/seq/veliev_e_sobel_operator/perf_tests/main.cpp
Comment (ev1ven) [reply tasks/seq/veliev_e_sobel_operator/perf_tests/main.cpp:1]: fixed

**Review by qfrff — APPROVED** [review]

**Review by Sedova-Olga — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

**Pull request merged**